src/util.ts

```diff
@@ -738,7 +738,7 @@ export function isNullOrUndefined(x?: any): x is null | undefined {
     return (x === null || x === undefined);
 }
 
-export function isWorkspaceFolder(x?: any): boolean {
+export function isWorkspaceFolder(x?: any): x is vscode.WorkspaceFolder {
```

Comment:
Instead of this, should we leave it as boolean and simply add a check in the method itself for `x is vscode.WorkspaceFolder`?

Reply:
If this is possible I think I'd rather keep the generic `boolean`

Reply:
It's returning a boolean. The difference is when `x is vscode.WorkspaceFolder` is used as return type, it lets the interpreter know its type when it returns true. I.E, when `x is vscode.WorkspaceFolder` is returned:
```
if (isWorkspaceFolder(x)) {
    // x is now a WorkspaceFolder
}
```

When returning boolean,
```
if (isWorkspaceFolder(x)) {
    // x is still any
}
```

Reply:
Ah, that sounds great! Thanks for the explanation.